### PR TITLE
Arg<Pat>

### PR DESCRIPTION
**Pull request by HoshinoTented**
Some problems:
- There are too many `.map(Arg::term)`
- Also `Arg<SerPat>` ?

### CODE REVIEW

File: base/src/main/java/org/aya/distill/CoreDistiller.java [hunk 1]
```diff
@@ -259,25 +259,29 @@ private ImmutableSeq<Arg<Term>> visibleArgsOf(Callable call) {
...
           options.map.get(AyaDistillerOptions.Key.ShowImplicitPats));
-        yield ctorDoc(outer, ctor.explicit(), ctorDoc, ctor.params().isEmpty());
+        yield ctorDoc(outer, licit, ctorDoc, ctor.params().isEmpty());
       }
-      case Pat.Absurd absurd -> Doc.bracedUnless(Doc.styled(KEYWORD, "()"), absurd.explicit());
-      case Pat.Tuple tuple -> Doc.licit(tuple.explicit(),
-        Doc.commaList(tuple.pats().view().map(sub -> pat(sub, Outer.Free))));
+      case Pat.Absurd absurd -> Doc.bracedUnless(Doc.styled(KEYWORD, "()"), licit);
+      case Pat.Tuple tuple -> Doc.licit(licit,
+        Doc.commaList(tuple.pats().view().map(sub -> pat(sub, true, Outer.Free))));   // TODO: Always true?
```
Comment (ice1000) [inline base/src/main/java/org/aya/distill/CoreDistiller.java:280]: Not always true I believe

Comment (HoshinoTented) [reply base/src/main/java/org/aya/distill/CoreDistiller.java:280]: `({a}, b)` is possible?

Comment (ice1000) [reply base/src/main/java/org/aya/distill/CoreDistiller.java]: I handled this case. I think this is yet impossible from the concrete but possible from elaboration.

File: base/src/main/java/org/aya/core/pat/Pat.java [hunk 2]
```diff
@@ -205,16 +194,17 @@ record ShapedInt(
...
 
     @Override public @NotNull Pat makeZero(@NotNull CtorDef zero) {
-      return new Pat.Ctor(explicit, zero.ref, ImmutableSeq.empty(), type);
+      return new Pat.Ctor(zero.ref, ImmutableSeq.empty(), type);
     }
 
     @Override public @NotNull Pat makeSuc(@NotNull CtorDef suc, @NotNull Arg<Pat> pat) {
       // TODO[ice]: Arg<Pat> in core
-      return new Pat.Ctor(explicit, suc.ref, ImmutableSeq.of(pat.term()), type);
+      // TODO[hsn]: This is my work.
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/pat/Pat.java:202]: Resolve?

File: base/src/main/java/org/aya/core/pat/PatToTerm.java [hunk 3]
```diff
@@ -34,7 +34,8 @@ public Term visit(@NotNull Pat pat) {
     var core = ctor.ref().core;
     var tele = core.selfTele;
     var args = ctor.params().zipView(tele)
-      .map(p -> new Arg<>(visit(p._1), p._2.explicit()))
+      // TODO: Is it true that `p._1.explicit = p._2.explicit` ?
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/pat/PatToTerm.java:37]: Can you check this for me?

**Review by ice1000 — COMMENTED** [review]
/cc @HoshinoTented

File: base/src/main/java/org/aya/core/pat/Pat.java (hunk 2 above)
Comment (HoshinoTented) [reply base/src/main/java/org/aya/core/pat/Pat.java:202]: Resolved.

Comment (ice1000) [reply base/src/main/java/org/aya/core/pat/Pat.java:202]: Can you delete the comment then?

**Review by ice1000 — APPROVED** [review]

File: base/src/main/java/org/aya/core/pat/PatToTerm.java (hunk 3 above)
Comment (HoshinoTented) [reply base/src/main/java/org/aya/core/pat/PatToTerm.java]: I believe it is true.

**Pull request merged**